src/app/CommandHandler.cpp

```diff
@@ -536,10 +639,14 @@ CHIP_ERROR CommandHandler::FinishCommand(bool aStartDataStruct)
     {
         ReturnErrorOnFailure(commandData.GetWriter()->EndContainer(mDataElementContainerType));
     }
+
+    if (mRefForResponse.HasValue())
```

Comment:
I don't think it can be done without breaking legacy. There might be code outside of the SDK that calls `CommandHandler::PrepareCommand` if I understand correctly. Which is why I am stuck saving this

Reply:
Please just file a followup issue to track that.  The answer might be separate legacy and non-legacy finish methods or something.  The legacy one will never have a ref...

Reply:
Added TODO comment on where we are setting the value in Prepare